apps/small_filters/Makefile

```diff
@@ -0,0 +1,157 @@
+include ../support/Makefile.inc
+
+
+# This app requires a separate toolchain to be built from the Android NDK,
+# using the make-standalone-toolchain.sh script:
+#$ build/tools/make-standalone-toolchain.sh --arch=arm64 --platform=android-21 --install-dir=$ANDROID_ARM64_TOOLCHAIN
+#$ build/tools/make-standalone-toolchain.sh --arch=arm --platform=android-21 --install-dir=$ANDROID_ARM_TOOLCHAIN
+CXX-host ?= $(CXX)
+CXX-arm-64-android ?= $(ANDROID_ARM64_TOOLCHAIN)/bin/aarch64-linux-android-c++
+CXX-arm-32-android ?= $(ANDROID_ARM_TOOLCHAIN)/bin/arm-linux-androideabi-c++
+CXX-arm-64-profile-android ?= $(CXX-arm-64-android)
+CXX-arm-32-profile-android ?= $(CXX-arm-32-android)
+
+CXXFLAGS-host ?=
+CXXFLAGS-arm-64-android ?=
+CXXFLAGS-arm-32-android ?=
+
+LDFLAGS-host ?= -lpthread -ldl -lm
+LDFLAGS-arm-64-android ?= -llog -fPIE -pie
+LDFLAGS-arm-32-android ?= -llog -fPIE -pie
+LDFLAGS-arm-64-profile-android ?= -llog -fPIE -pie
+LDFLAGS-arm-32-profile-android ?= -llog -fPIE -pie
+
+BIN ?= bin
+
+FILTERS ?= conv3x3a16 dilate3x3 median3x3 gaussian5x5 sobel conv3x3a32
+
+OBJS = $(patsubst %,$(BIN)/\%/%_hvx64.o, $(FILTERS))
+OBJS += $(patsubst %,$(BIN)/\%/%_hvx128.o, $(FILTERS))
+OBJS += $(patsubst %,$(BIN)/\%/%_cpu.o, $(FILTERS))
+
+LINK_OBJS = $(patsubst %,$(BIN)/$$*/%_hvx64.o, $(FILTERS))
+LINK_OBJS += $(patsubst %,$(BIN)/$$*/%_hvx128.o, $(FILTERS))
+LINK_OBJS += $(patsubst %,$(BIN)/$$*/%_cpu.o, $(FILTERS))
+
+UPPERCASE_FILTERS = $(shell echo $(FILTERS) | tr '[:lower:]' '[:upper:]')
+DASH_D_DEFINES = $(patsubst %, -D%=1, $(UPPERCASE_FILTERS))
+
+
+$(BIN)/conv3x3a16: conv3x3a16.cpp $(GENERATOR_DEPS)
+	@-mkdir -p $(BIN)
+	echo "${GENERATOR_DEPS}"
+	$(CXX) $(CXXFLAGS) -O3 -g -fno-rtti $(filter-out %.h,$^) -o $@ $(LDFLAGS)
+
+$(BIN)/%/conv3x3a16_cpu.o: $(BIN)/conv3x3a16
```

Comment:
Yes, this is my biggest gripe with this Makefile. I tried very hard to do exactly what you have in mind but eventually gave up and had to go down the yucky path of enumerating 4 rules per filter.